diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4537cdd..8265ddc 100644
--- a/src/protobuf/CMakeLists.txt
+++ b/src/protobuf/CMakeLists.txt
@@ -24,7 +24,7 @@ if(protobuf_DEPRECATED_CMAKE_SUBDIRECTORY_USAGE)
 endif()
 
 # Options
-option(protobuf_INSTALL "Install protobuf binaries and files" ON)
+option(protobuf_INSTALL "Install protobuf binaries and files" OFF)
 option(protobuf_BUILD_TESTS "Build tests" ON)
 option(protobuf_BUILD_CONFORMANCE "Build conformance tests" OFF)
 option(protobuf_BUILD_EXAMPLES "Build examples" OFF)
@@ -279,7 +279,8 @@ endif (MSVC)
 include_directories(
   ${ZLIB_INCLUDE_DIRECTORIES}
   ${protobuf_BINARY_DIR}
-  ${protobuf_SOURCE_DIR}/src)
+  ${protobuf_SOURCE_DIR}/src
+  ${protobuf_SOURCE_DIR}/config)
 
 set(protobuf_ABSL_PROVIDER "module" CACHE STRING "Provider of absl library")
 set_property(CACHE protobuf_ABSL_PROVIDER PROPERTY STRINGS "module" "package")
diff --git a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
index f343458..d881215 100644
--- a/src/protobuf/cmake/libprotobuf-lite.cmake
+++ b/src/protobuf/cmake/libprotobuf-lite.cmake
@@ -21,7 +21,7 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf-lite PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf-lite PRIVATE log)
+  target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/src)
 target_link_libraries(libprotobuf-lite PUBLIC ${protobuf_ABSL_USED_TARGETS})
diff --git a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
index 422754a..04f8449 100644
--- a/src/protobuf/cmake/libprotobuf.cmake
+++ b/src/protobuf/cmake/libprotobuf.cmake
@@ -24,7 +24,7 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf PRIVATE log)
+   target_include_directories(libprotobuf PRIVATE ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
 target_link_libraries(libprotobuf PUBLIC ${protobuf_ABSL_USED_TARGETS})
